lib/ASTGen/CMakeLists.txt

```diff
@@ -38,3 +38,31 @@ add_pure_swift_host_library(swiftASTGen STATIC
     SwiftSyntaxMacroExpansion
     swiftLLVMJSON
 )
+
+set(c_include_paths
+  # LLVM modules and headers.
+  "${LLVM_MAIN_INCLUDE_DIR}"
+  # Generated LLVM headers.
+  "${LLVM_INCLUDE_DIR}"
+  # Clang modules and headers.
+  ${CLANG_INCLUDE_DIRS}
+  # Bridging modules and headers.
+  "${SWIFT_MAIN_INCLUDE_DIR}"
+  # Generated C headers.
+  "${CMAKE_CURRENT_BINARY_DIR}/../../include")
+set(c_include_paths_args)
+foreach(c_include_path ${c_include_paths})
+  list(APPEND c_include_paths_args "SHELL: -Xcc -I -Xcc ${c_include_path}")
+endforeach()
+
+set(compile_options
+  ${c_include_paths_args}
+  "SHELL: -Xfrontend -enable-experimental-cxx-interop"
```

Comment:
is it possible to use the `-cxx-interoperability-mode=default` flag here instead?

Reply:
the existing experimental flag is deprecated

Reply:
I believe that flag was added with 5.9, so we would need to require a 5.9 host compiler for that (which would require updating CI first)

Reply:
I see, got it.

Reply:
Ended up making this conditional on 5.9, so we will use the new flag where we can